pkg/chunked/compressor/compressor.go

```diff
@@ -420,6 +420,14 @@ func writeZstdChunkedStream(destFile io.Writer, outMetadata map[string]string, r
 		zstdWriter.Close()
 		return err
 	}
+
+	// make sure the entire tarball is flushed to the output as it might contain
+	// some trailing zeros that affect the checksum.
+	if _, err := io.Copy(zstdWriter, its); err != nil {
```

Comment:
For the record: This looked very surprising to me (if notthing else, I doubted that after passing `its` to `tar.NewReader` there are any promises at all about the position of `its`)…

But tar-split internally does the same thing in https://github.com/vbatts/tar-split/blob/9a95e026024d3929928021244d72e59f758616b3/tar/asm/disassemble.go#L121-L151 . 

(And for the purposes of this PR, we don’t care whether the unexpected trailing padding is recorded into tar-split metadata (that _is_ important, but hopefully ensured by existing code), but whether we write it into the `zstdWriter` unmodified; so we must do that separately.)